src/main/java/io/rainfall/statistics/monitor/MemStatisticsCollector.java

```diff
@@ -0,0 +1,115 @@
+package io.rainfall.statistics.monitor;
+
+import io.rainfall.reporting.HtmlReporter;
+import io.rainfall.statistics.collector.StatisticsCollector;
+import io.rainfall.statistics.exporter.Exporter;
+import io.rainfall.statistics.exporter.HtmlExporter;
+import io.rainfall.statistics.exporter.TextExporter;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.Writer;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * Created by KECL on 8/18/2016.
+ */
+public class MemStatisticsCollector  implements StatisticsCollector {
+
+  private static final MemoryMXBean MEM_BEAN = ManagementFactory.getMemoryMXBean();
+  private static final String MEM_STATS = "Memory Utilization";
+
+  private Writer memOutput;
+
+  public static class MemStats {
```

Comment:
Actually since a MemStatisticsCollector instances will have only 1 MemStats instance, you can probably even get rid of this class and just store directly in MemStatisticsCollector  the two fields startTimestamp and memUtilization